### ISSUE TITLE
error on secure route after authenticated  from mongoldb

### ISSUE THREAD

**Issue by sivi29y** [ParsePlatform/parse-server-example#191]
in var/log/mongodb
 [conn122] Successfully authenticated as principal parse on  ... 

yet parse server query replies with 
{"code":1,"message":"Internal server error."}

and node 

'server  myserrver.com:27017 received an error {"name":"MongoError","message":"write EPIPE"}'

I don't think it's implementation problem it's looks like a bug in the parse client. Any help is appreciated

**Comment by drew-gross** [comment]
`MongoError` means this issue is in your database configuration. For help with issues with that, please use Stack Overflow or Server Fault.

**Comment by sivi29y** [comment]
How come the problem is on  mongodb if i can connect to mongodb from the mongo shell?. I'm not sure the problem is at mongodb at all

**Comment by drew-gross** [comment]
There are other types of error besides "Unable to connect" error. This appears to be a "write EPIPE" error. Which could have many causes. Try turning `VERBOSE=1` on for your Parse Server to see verbose logs.

**Comment by rahimrahman** [comment]
@sivi29y did you figure out your issue?  I think I have similar problem.

**Comment by rahimrahman** [comment]
@sivi29y I've figured it out.  And I'm posting this just in case somebody else is having the same issue where parse-server is not able to connect to mongo db after SSL has been enabled on that mongo instance.  What I had to do was to make sure that my mongod instance would start with `--sslAllowConnectionsWithoutCertificates` option.

I'm using docker, so I spun another mongo container and tried to connect to my SSLed mongo instance on a different container:

```
$ mongo --ssl --host mymongodb
Failed global initialization: BadValue need to either provide sslCAFile or specify sslAllowInvalidCertificates
```

Since this separate mongo container doesn't have the .pem files I created for the SSL mongo instance, I figured I had to enable `sslAllowInvalidCertificates` option.

Hope this helps somebody.
